Lesson2/src/Homework2.java

```diff
@@ -0,0 +1,112 @@
+import java.util.Scanner;
+
+import static java.lang.Math.sqrt;
+
+public class Homework2 {
+
+    public static void main(String[] args) {
+
+        System.out.println(sum(10, 20));
+        System.out.println(sum(Integer.MAX_VALUE, Integer.MAX_VALUE));
+        System.out.println(max(50, 49));
+        System.out.println(average(new int[]{1, 2, 3, 4, 5}));
+        System.out.println(max(new int[]{1, 2, 3, 4, 5, 100, 99}));
+        System.out.println(calculateHypotenuse());
+    }
+
+    /**
+     * 1. Method should return sum of "a" and "b"
+     * 2. (optional) if sum of a and b is more then int can store -- return -1;
+     **/
+    public static int sum(int a, int b) {
+        int x = a + b;
+        int z = Integer.MAX_VALUE + Integer.MAX_VALUE;
```

Comment:
this will not work, unfortunately. value of z will be equal to -2, because int can't store such big numbers. You need to change z variable type to long and make sure that the expression "Integer.MAX_VALUE + Integer.MAX_VALUE" will be cast to long

Please let me know if you understand my comment